plugin/output/elasticsearch/elasticsearch.go

```diff
@@ -52,6 +59,31 @@ type Config struct {
 	//> The list of elasticsearch endpoints in the following format: `SCHEMA://HOST:PORT`
 	Endpoints []string `json:"endpoints"  required:"true"` //*
 
+	//> @3@4@5@6
+	//>
+	//> Elastic username for Basic authentication (username and password).
+	Username string `json:"username"` //*
+
+	//> @3@4@5@6
+	//>
+	//> Elastic password for Basic authentication (username and password).
+	Password string `json:"password"` //*
+
+	//> @3@4@5@6
+	//>
+	//> Elastic secret token for API key authentication.
+	APIKey string `json:"api_key"` //*
+
+	//> @3@4@5@6
+	//>
+	//> Custom certificate authority used to sign the certificates of cluster nodes.
```

Comment:
This cannot be exactly copied, because there is about an array of bytes, and we have a path to a file. Please, resolve this discussion if the new version is OK :)